src/main/java/com/microsoft/jenkins/kubernetes/command/DeploymentCommand.java

```diff
@@ -143,6 +103,118 @@ String getMasterHost(KubernetesClientWrapper wrapper) {
         return "Unknown";
     }
 
+    static class DeploymentTask extends MasterToSlaveCallable<TaskResult, AbortException> {
+        private FilePath workspace;
+        private TaskListener taskListener;
+        private ClientWrapperFactory clientFactory;
+        private EnvVars envVars;
+
+        private String configPaths;
+        private String secretNamespace;
+        private String secretNameCfg;
+        private String defaultSecretNameSeed;
+        private boolean enableSubstitution;
+
+        private List<ResolvedDockerRegistryEndpoint> dockerRegistryEndpoints;
+
+        @Override
+        public TaskResult call() throws AbortException {
+            try {
+                return doCall();
+            } catch (Exception ex) {
+                // JENKINS-50760
+                // JEP-200 restricts the classes allowed to be serialized with XStream to a whitelist.
+                // The task being executed in doCall may throw some exceptions from the third party libraries,
+                // which will cause SecurityException when it's transferred from the slave back to the master.
+                // We catch the exception and print the stacktrace here (slave), and throw an exception which can
+                // be serialized properly.
+                ex.printStackTrace(taskListener.error(ex.getMessage()));
+                throw new AbortException(ex.getMessage());
```

Comment:
Nice2Have: Add a text suggesting checking agent logs for more information

Reply:
At noted in the previous comment, we don't need further notice. Better to update it to `ProxyException` as @jglick mentioned. I will update that.

Reply:
updated.